File: leetcode/js/53-maximum-subarray(E).js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var maxSubArray = function(nums) {

  let totalMax = nums[0]
  let currentMax = 0;
  
  
  for (const number of nums) {
      currentMax += number
      
      if (currentMax < number) {
          currentMax = number
      }
      
      totalMax = Math.max(totalMax, currentMax)
      
  }
  
  return totalMax
  
};

File: leetcode/python/53-maximum-subarray(E).py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = nums[0]
        
        currentSum = 0
        for i, a in enumerate(nums):
            
            if currentSum + a > a:
                currentSum += a
            else:
                currentSum = a
            if currentSum > maxSum:
                maxSum = currentSum
            
        return maxSum